### ISSUE TITLE
[napari-workflows] Access parameters within napari-workflow?

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#158]
placeholder, I'll update later (e.g. True,True,True,True, in measurement for 3D)

**Comment by jluethi** [comment]
If this is about https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/152, I think that's mostly out-of-scope for Fractal. The user should provide pipelines with correct settings for the resolution levels they want to run things at. We can't know which parameters would need to be scaled when.

If it's about changing other parameters, I'm happy to have the discussion whether those should be modified within Fractal or whether that should be done outside of Fractal :)

**Comment by tcompa** [comment]
TL;DR
This issue was not specially relevant, and can be closed right away. I thought we'd have to validate some inputs (the `perimeter=True/False` one) to avoid unsafe execution, but actually the napari-workflow function (a wrapper of scikit-image itself) already does that for us.
If access to workflow parameters comes up somewhere else we can open a new issue. I guess the typical answer will be "user is in charge".

---

The `regionprops` function that we use as part of workflows has [these arguments](https://github.com/haesleinhuepf/napari-skimage-regionprops/blob/master/napari_skimage_regionprops/_regionprops.py#L11):
```python
def regionprops(
  image_layer : napari.layers.Layer,
  labels_layer: napari.layers.Labels,
  size : bool = True,
  intensity : bool = True,
  perimeter : bool = False,
  shape : bool = False,
  position : bool = False,
  moments : bool = False,
  napari_viewer : Viewer = None):
```

On one example (I don't have a reproducible script at hand, but I can recreate it if is useful) running on 3D data and with all arguments set to `True` led to a memory error (asking for >50G of memory allocation, for the tiny dataset). This seemed weird and I switched to `True, True, False, False, False, False`, which was sufficient to avoid the error.

At first (and this is the reason for the current issue) I thought that the problem was that perimeter measurement is not allowed for 3D images. Then I realized that the napari-workflows function linked above actually raises a warning about that, and then skips this property. Therefore the large memory usage should have come from some other property (`shape`, `position` or `moments`). I did not dig further to identify it.

**Comment by jluethi** [comment]
Ok, makes sense, thanks for the context! :) They work well on 2D examples, but some may not make sense for 3D measurements :)
